lib/MetaCPAN/Model/Release.pm

```diff
@@ -271,6 +273,34 @@ sub _build_files {
     return \@files;
 }
 
+sub set_changes_file {
+    my $self  = shift;
+    my @files = @{ $_[0] };
+
+    my @changes_files = qw(
+        Changelog
+        ChangeLog
+        CHANGELOG
+        Changes
+        CHANGES
+        NEWS
+    );
+
+    if ( $files[0]->distribution eq 'perl' ) {
+        foreach my $file (@files) {
+            if ( $file->name eq 'perldelta.pod' ) {
+                return $file->path;
+            }
+
+        }
+    }
+
+    foreach my $file (@files) {
+        return $file->path if grep { $_ eq $file->name } @changes_files;
```

Comment:
Good catch.  I think just in the top level, with the perl releases being the exception.  Adding a test case is a good idea.